reddit/parse.go

```diff
@@ -67,6 +68,37 @@ func (p *parserImpl) parse(
 	)
 }
 
+// parse_submitted parses a response from reddit describing
+// the status of some resource that was submitted
+func (p *parserImpl) parse_submitted(blob json.RawMessage) (Submission, error) {
+	var wrapped map[string]interface{}
+	err := json.Unmarshal(blob, &wrapped)
+	if err != nil {
+		return Submission{}, err
+	}
+
+	wrapped = wrapped["json"].(map[string]interface{})
+	if len(wrapped["errors"].([]interface{})) != 0 {
+		return Submission{}, fmt.Errorf("API errors were returned: %v", wrapped["errors"])
+	}
+
+	data := wrapped["data"].(map[string]interface{})
+
+	// Comment submissions are further wrapped in a things block
+	// because of ... something? There only appears to be a single thing
+	// This transformes var data to be the data of the single thing
+	// This also mirrors the key permalink -> url because they are functionally the same
```

Comment:
Does permalink include the scheme and domain, or is just the path? I would expect something called "url" to contain the scheme and domain.

Reply:
Yes, I missed that. That block has been updated to prefix the permalink, though I wonder if there's a better way that I should be going about that

Reply:
I think your solution is fine.

If you squash this commit into `Create parse_submitted` I will merge.

Reply:
History was updated